lib/Controller/FreePromptController.php

```diff
@@ -33,9 +33,9 @@ public function __construct(
 	 */
 	#[NoAdminRequired]
 	#[NoCSRFRequired]
-	public function processPrompt(string $prompt, int $nResults = 1): DataResponse {
+	public function processPrompt(string $prompt): DataResponse {
 		try {
-			$result = $this->freePromptService->processPrompt($prompt, $nResults);
+			$result = $this->freePromptService->processPrompt($prompt);
```

Comment:
The logic of checking the userId should be moved from `FreePromptService` to `FreePromptController`. The controller checks if it's valid (is null in our case) and, if so, calls the service method passing the userId as param, if not, returns an error response.

The `$userId` param of service methods can have the `string` type (not `?string`) if you know this action only makes sense for an authenticated user and the userId has been checked in the controller. This applies to all controllers and services.

Reply:
Ah, yeah, I forgot to do that!